--- SDK/include/c_types.h	2021-10-15 11:40:39.000000000 +0000
+++ BUILD/c_types.h	2021-12-12 23:59:14.000000000 +0000
@@ -25,6 +25,8 @@
 #ifndef _C_TYPES_H_
 #define _C_TYPES_H_
 
+#ifndef C2NIM
+
 typedef unsigned char       uint8_t;
 typedef signed char         sint8_t;
 typedef signed char         int8_t;
@@ -75,6 +77,8 @@
 #define NULL (void *)0
 #endif /* NULL */
 
+#endif /* C2NIM */
+
 /* probably should not put STATUS here */
 typedef enum {
     OK = 0,
@@ -84,11 +88,14 @@
     CANCEL,
 } STATUS;
 
+
 #define BIT(nr)                 (1UL << (nr))
 
 #define REG_SET_BIT(_r, _b)  (*(volatile uint32_t*)(_r) |= (_b))
 #define REG_CLR_BIT(_r, _b)  (*(volatile uint32_t*)(_r) &= ~(_b))
 
+#ifndef C2NIM
+
 #define DMEM_ATTR __attribute__((section(".bss")))
 #define SHMEM_ATTR
 
@@ -102,6 +109,8 @@
 
 #define STORE_ATTR __attribute__((aligned(4)))
 
+#include <stdbool.h>
+#if 0
 #ifndef __cplusplus
 typedef unsigned char   bool;
 #define BOOL            bool
@@ -109,8 +118,10 @@
 #define false           (0)
 #define TRUE            true
 #define FALSE           false
-
+#endif
 
 #endif /* !__cplusplus */
 
+#endif /* C2NIM */
+
 #endif /* _C_TYPES_H_ */
