--- SDK/include/os_type.h	2018-02-06 12:45:58.000000000 +0000
+++ BUILD/os_type.h	2019-08-22 08:26:00.000000000 +0000
@@ -25,6 +25,19 @@
 #ifndef _OS_TYPES_H_
 #define _OS_TYPES_H_
 
+#ifdef C2NIM
+#@
+type
+  os_signal_t* {.importc: "os_signal_t", header: "os_type.h", nodecl.} = object
+  os_param_t* {.importc: "os_param_t", header: "os_type.h", nodecl.} = object
+  os_timer_t* {.importc: "os_timer_t", header: "os_type.h", nodecl.} = object
+  os_event_t* {.importc: "os_event_t", header: "os_type.h", nodecl.} = object
+  os_task_t* {.importc: "os_task_t", header: "os_type.h", nodecl.} = object
+  os_timer_func_t* = proc(arg: pointer) {. cdecl .} # See: http://forum.nim-lang.org/t/1801
+                                             # http://nim-lang.org/docs/manual.html#types-procedural-type
+@#
+#else /* C2NIM */
+
 #include "ets_sys.h"
 
 #define os_signal_t ETSSignal
@@ -34,4 +47,6 @@
 #define os_timer_t  ETSTimer
 #define os_timer_func_t ETSTimerFunc
 
+#endif /* C2NIM */
+
 #endif
