--- SDK/include/upgrade.h	2018-02-06 12:45:58.000000000 +0000
+++ BUILD/upgrade.h	2019-08-22 08:26:00.000000000 +0000
@@ -55,12 +55,20 @@
     uint8 *url;
 
     upgrade_states_check_callback check_cb;
-    struct espconn *pespconn;
+    // Wye?
+    //#ifndef C2NIM
+    //struct espconn *pespconn;
+    //#else
+    void *pespconn;
+    //#endif
 };
+typedef struct upgrade_server_info upgrade_server_info; // ESP SDK
 
+/*
 #define UPGRADE_FLAG_IDLE       0x00
 #define UPGRADE_FLAG_START      0x01
 #define UPGRADE_FLAG_FINISH     0x02
+*/
 
 void system_upgrade_init();
 void system_upgrade_deinit();
